material_3_demo/lib/main.dart

```diff
@@ -0,0 +1,724 @@
+import 'package:flutter/material.dart';
+import 'package:material_3_demo/surface_tones_screen.dart';
+import 'package:material_3_demo/typography_screen.dart';
+
+import 'color_palettes_screen.dart';
+
+void main() {
+  runApp(const Material3Demo());
+}
+
+class Material3Demo extends StatefulWidget {
+  const Material3Demo({Key? key}) : super(key: key);
+
+  @override
+  State<Material3Demo> createState() => _Material3DemoState();
+}
+
+const Color m3BaseColor = Color(0xff6750a4);
+const List<Color> colorOptions = [
+  m3BaseColor,
+  Colors.blue,
+  Colors.teal,
+  Colors.green,
+  Colors.yellow,
+  Colors.orange,
+  Colors.pink
+];
+const List colorText = [
+  "M3 Base Color",
+  "Blue",
+  "Teal",
+  "Green",
+  "Yellow",
+  "Orange",
+  "Pink",
+];
```

Comment:
Thanks a lot for the suggestions! We also thought about a color picker before, but to implement that, we will need a third-party package, so not sure if it is a suitable solution to show a M3 demo with some plugin code. So currently, we just provided a single color as `colorSchemeSeed` and to generate the theme. 

As for the first solution, it looks cool! How do you think? @darrenaustin

Reply:
Yeah, as Qun pointed out this was mostly to avoid depending on a third-party package for a color chooser (or building one just for this demo). We still wanted to show what different color schemes looked like on the components though.

The shuffle color button is interesting. Is it just picking a totally random seed color, or is it choosing a random entry from a set of seed colors?

We could do either pretty easily. If we do, it might still be nice to have the list of preset colors as well. Perhaps we could add a new button to the AppBar that does the shuffle, but leave the menu button to choose one from the predefined set we have.